### ISSUE TITLE
ERROR: Cannot convert the "System.Object[]" value of type "System.Object[]" to type "System.Version". The install of microsoft-ui-xaml was NOT successful.

### ISSUE THREAD

**Issue by Axer128** [chocolatey/choco#3247]
### Checklist

- [X] I have verified this is the correct repository for opening this issue.
- [X] I have verified no other issues exist related to my problem.
- [X] I have verified this is not an issue for a specific package.
- [X] I have verified this issue is not security related.

### What You Are Seeing?

Fresh install of choco 2.1 on windows 11.
I first did choco install GoogleChrome - this worked fine.
Than:
choco install winget
Chocolatey v2.1.0
Installing the following packages:
winget
By installing, you accept licenses for the packages.
Progress: Downloading microsoft-ui-xaml 2.7.0... 100%

microsoft-ui-xaml v2.7.0 [Approved]
microsoft-ui-xaml package files install completed. Performing other installation steps.
The package microsoft-ui-xaml wants to run 'chocolateyinstall.ps1'.
Note: If you don't run this script, the installation will fail.
Note: To confirm automatically next time, use '-y' or consider:
choco feature enable -n allowGlobalConfirmation
Do you want to run the script?([Y]es/[A]ll - yes to all/[N]o/[P]rint): Y

ERROR: Cannot convert the "System.Object[]" value of type "System.Object[]" to type "System.Version".
The install of microsoft-ui-xaml was NOT successful.

Trying to install most any other package than results in the same error. eg: cpu-z:
- cpu-z - Unable to resolve dependency 'microsoft-ui-xaml'.

Trying to directly install microsoft-ui-xaml i get this error:
Unable to resolve dependency 'microsoft-vclibs'

(all versions of ms vc redists are installed on my system and working fine)

Trying to directly install microsoft-vclibs i get this error:
Unable to resolve dependency 'microsoft-ui-xaml

So circular dependency issue, and the one in the title. And it seems attempting to install winget has broken my chocolatey completely.

### What is Expected?

It installs winget without issue.. And shouldn't complain about those dependencies, because they are already installed and functional on my system - as a manual install of winget worked fine.

### How Did You Get This To Happen?

I put this all in the first box.

### System Details

* Operating System: Windows 11 22H
* Windows PowerShell version: 
PSVersion                      7.2.5
PSEdition                      Core
GitCommitId                    7.2.5
OS                             Microsoft Windows 10.0.22621
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
* Chocolatey CLI Version: 2.1.0
* Chocolatey Licensed Extension version: Chocolatey v2.1.0
* Chocolatey License type: Personal
* Terminal/Emulator: Powershell 7

### Installed Packages

```bash
chocolatey 2.1.0
chocolatey-compatibility.extension 1.0.0
chocolatey-core.extension 1.4.0
GoogleChrome 114.0.5735.199
winget 1.4.11071
winget-cli 1.4.11071
6 packages installed.
```

### Output Log

```bash
https://gist.github.com/Axer128/7abc448296221a3a1a376463fd1e1a21
```

### Additional Context

Had the same issue with choco 2.0 , so uninstalled fully deleting everything, installed clean via command line provided on choco website and error persists.

**Comment by pauby** [comment]
The winget package issue I _think_ I know what the issue is. I'm the maintainer of that package.

I'll raise an issue over of my repository for this.

Does the issue only occur, with other packages, if you install the winget package first? If you don't install the winget package, do other packages install without problems?

**Comment by Axer128** [comment]
It may not purely be the winget package. I think it's ANY package that wants the microsoft-ui-xaml as a prereq, which is a lot.. Often incorrectly - as CPU-Z works fine without that package installed. I did some digging into the script to see why it was failing, and the error I get if I manually install it is that a newer version is already installed (which yea it is, the version choco downloads is outdated)

As yea I also tried a clean install for CPU-Z and same thing, it fails on the microsoft-ui-xaml with the same system.object error first.

I believe the same error happens with the vclibs-package. 
So yea choco should manually be seeing those are installed, not enforcing potentially outdated versions which fail and break many installs.

And my choco isn't completely broke, just any packages that depend on those decencies it seems. Some packages without either of those prereqs still install fine. EG: GPU-Z installs fine.

In the meantime is there a hack I can do to tell choco that both of these prereqs are fully up to date and stop bothering me over them?

**Comment by pauby** [comment]
I'm fairly certain this is a Windows 11 only issue so shouldn't affect Windows 10 people.

> So yea choco should manually be seeing those are installed, especially vclibs since there is a massively high chance the user already has recent versions of it installed before they install choco.. not enforcing potentially outdated versions which fail and break many installs.

Just to be clear, that's not Chocolatey CLI's responsibility. That's the responsibility of the package.

> In the meantime is there a hack I can do to tell choco that both of these prereqs are fully up to date and stop bothering me over them?

Install the packages manually and add `--skip-powershell` to the command line. 

In the meantime I'm going to close this as it's a package issue and not a Chocolatey CLI one. I'll raise an issue over on my repository later today and tag you in that.

**Comment by Axer128** [comment]
Can't install teh packages because as I said, they are already installed and newer version that are on chocolatey and the appX package adder refuses to install old versions when newer present. So should this issue be moved to microsoft-ui-xaml repo? (It's very outdated, wants 2.7 but 2.84+ is available and windows 11 or something related on my system auto installed a newer version required - I didn't update it manually)

Only way I could do as you suggest would be to uninstall them, then manually install the old versions choco wants. Which could break a lot of my other applications that depend on them so not a viable option.

**Comment by pauby** [comment]
> Can't install teh packages because as I said, they are already installed and newer version that are on chocolatey and the appX package adder refuses to install old versions when newer present.

Did you run the install with the added option I provided?

**Comment by Axer128** [comment]
I did try that but it would still give me the endless cyclic dependency issue - xaml wanted vclibs, vclibs wanted xaml.. So I deleted my choco dir yet again and installed fresh than ran installed each manually with the --skip-powershell. And it worked yay.

I was able than than install CPU-Z without issues , it grabbed a couple other dependencies it also didnt need and those installed fine.

Winget however.. Does now install fine, but doesn't actually install anything. No winget.exe anywhere on my system. The C:\ProgramData\chocolatey\lib\winget dir just has winget.nupkg and winget.nuspec
